Server/src/transport/unity_instance_middleware.py

```
@@ -260,3 +297,108 @@ async def on_read_resource(self, context: MiddlewareContext, call_next):
+        context: MiddlewareContext,
+    ) -> set[str] | None:
+        ctx = context.fastmcp_context
+        user_id = ctx.get_state("user_id") if config.http_remote_hosted else None
+        active_instance = ctx.get_state("unity_instance")
+
+        project_hashes = self._resolve_candidate_project_hashes(active_instance)
+        if not project_hashes:
+            try:
+                sessions_data = await PluginHub.get_sessions(user_id=user_id)
+                sessions = sessions_data.sessions if sessions_data else {}
+            except Exception:
```

Ruff (0.15.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)